pkg/controllermanager/controller/secretbinding/secretbinding_control.go

```diff
@@ -163,6 +175,33 @@ func (r *secretBindingReconciler) Reconcile(ctx context.Context, request reconci
 		}
 	}
 
+	if len(secretBinding.Quotas) != 0 {
+		for _, objRef := range secretBinding.Quotas {
+			quota := &gardencorev1beta1.Quota{}
+			if err := r.gardenClient.Get(ctx, kutil.Key(objRef.Namespace, objRef.Name), quota); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Add 'referred by a secret binding' label
+			if !metav1.HasLabel(quota.ObjectMeta, v1beta1constants.LabelSecretBindingReference) {
+				patch := client.MergeFrom(quota.DeepCopy())
+				metav1.SetMetaDataLabel(&quota.ObjectMeta, v1beta1constants.LabelSecretBindingReference, "true")
+				if err := r.gardenClient.Patch(ctx, quota, patch); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to add referred label to the quota referenced in SecretBinding, quota: %s , namespace: %s : %w", quota.Name, quota.Namespace, err)
+				}
```

Comment:
This is always the same code. Does it make sense to extract this into a function

`reconcileReferenceLabel(context.Context, client.Object, bool)` where the third parameter is either `true` (-> add label) or `false` (-> remove label)?

Reply:
How do we take care of the two different objects, in the patch call? and also in adding the labels.
Are you proposing to use different `reconcileLabel` for secrets and quotas?